### PR TITLE
new post page

### PR DESCRIPTION
**Pull request by nm-le**
### Description

Create createpost.tsx inside apps/nextjs/src/pages to handle creating new pages. New posts are default to type 'announcement' for now, but will add option for user to select type of post (either announcement or discussion). 

Also during testing I can only create new post by changing createPost to use publicProcedure. This is reflected in the PR but will need to change this back to adminProcedure. Also added /createpost to publicRoutes in middleware.ts to bypass Clerk.

image
 

image

### Reason for Change

Fixes CFD-

Completes CFD-34

Contributes to CFD-

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other

### Testing

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: apps/nextjs/src/middleware.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { authMiddleware } from "@clerk/nextjs";
 export default authMiddleware({
   // "/" is not protected since that's where we land
   // api routes are not protected since authentication is done via protected routes in trpc.ts
-  publicRoutes: ["/", "/(api|trpc)(.*)"],
```
Comment (BakerWJ) [inline apps/nextjs/src/middleware.ts:7]: this page should not be public

File: package.json [hunk 2]
```diff
@@ -25,6 +25,8 @@
     "@cfd/prettier-config": "^0.1.0",
     "@manypkg/cli": "^0.21.0",
     "prettier": "^3.0.3",
+    "react-hook-form": "^7.48.2",
```
Comment (BakerWJ) [inline package.json:28]: this should be added to the package.json in the nextjs folder

File: package.json [hunk 3]
```diff
@@ -25,6 +25,8 @@
...
+    "react-query": "^3.39.3",
```
Comment (BakerWJ) [inline package.json:29]: delete

File: packages/api/src/router/announcement.ts [hunk 4]
```diff
@@ -11,7 +16,7 @@ export const announcementRouter = createTRPCRouter({
       where: { postType: "Announcement" },
     });
   }),
-  createAnnouncement: adminProcedure
+  createAnnouncement: publicProcedure
```
Comment (BakerWJ) [inline packages/api/src/router/announcement.ts:19]: revert

File: apps/nextjs/src/pages/createpost.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      border: "none",
+      backgroundColor: "#007bff",
+      color: "white",
+      cursor: "pointer",
+    },
+    error: {
+      color: "red",
+      fontSize: "0.8rem",
+    },
+  };
```
Comment (BakerWJ) [inline apps/nextjs/src/pages/createpost.tsx:77]: for now can just leave form unstyled

**Review by BakerWJ — CHANGES_REQUESTED** [review]
can you try logging into the website and then trying to access the form and submit?

**Comment by nm-le** [comment]
Hi, not sure how to fix failed checks after I made the changes to the package.json files. Submitting form works after logging in the website.

**Comment by BakerWJ** [comment]
Run pnpm i to fix the lockfile

**Review by BakerWJ — COMMENTED** [review]
looks good -- can you update the image in the PR description? will approve after addressing the one comment

File: apps/nextjs/package.json [hunk 6]
```diff
@@ -24,6 +24,8 @@
     "next": "13.4.12",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-hook-form": "^7.48.2",
+    "react-query": "^3.39.3",
```
[the remark this answers is not in the document]
Comment (nm-le) [reply apps/nextjs/package.json:28]: I thought you wanted that :?

image

[the remark this answers is not in the document]
Comment (BakerWJ) [reply apps/nextjs/package.json:28]: not sure why its highlighting both lines for this comment 🥲 -- comment is only about react-query

[the remark this answers is not in the document]
Comment (nm-le) [reply apps/nextjs/package.json:28]: oh ok makes sense

**Review by BakerWJ — APPROVED** [review]

**Pull request merged**